test/unit-test/ota_utest.c

```diff
@@ -1643,6 +1674,27 @@ void test_OTA_ReceiveFileBlockCompleteDynamicBufferHttp()
     test_OTA_ReceiveFileBlockCompleteHttp();
 }
 
+/* Test that extractAndStoreArray fails if device does not have sufficient
+ * memory to allocate the string/array. */
+void test_OTA_ExtractArrayMemAllocFails()
+{
+    otaInterfaces.os.mem.malloc = mockMallocAlwaysFail;
+
+    memset( &pOtaAppBuffer, 0, sizeof( pOtaAppBuffer ) );
```

Comment:
Are there specific fields in pOtaAppBuffer that you are trying to set to 0? I think changing this line to explicitly set those fields to 0 would be more descriptive towards what you are trying to test.

Reply:
Explicitly set streamNameSize to 0.